.circleci/config.yml

```diff
@@ -1,89 +1,114 @@
       - run:
-          name: Install Dependencies
-          command: NODE_ENV=dev yarn
+          name: Install dependencies
+          command: yarn --frozen-lockfile --non-interactive
+
+      # Save cache
       - save_cache:
-          key: yarn-{{ checksum "yarn.lock" }}
+          key: lock-{{ checksum "yarn.lock" }}
           paths:
-            - "node_modules"
+            - node_modules
+
+      # Persist workspace
       - persist_to_workspace:
           root: ~/project
           paths:
-            - bootstrap-vue
+            - node_modules
 
+  # --------------------------------------------------------------------------
+  # Phase 2: Lint + Audit + Build Nuxt and fixtures
+  # --------------------------------------------------------------------------
   lint:
     <<: *defaults
     steps:
+      - checkout
       - attach_workspace:
           at: ~/project
       - run:
           name: Lint
           command: yarn lint
 
+  audit:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Security audit
+          command: yarn audit
+
   build:
     <<: *defaults
     steps:
+      - checkout
       - attach_workspace:
           at: ~/project
       - run:
           name: Build
           command: yarn build
 
-  test:
+  # --------------------------------------------------------------------------
+  # Phase 3: Tests
+  # --------------------------------------------------------------------------
+  test-unit:
     <<: *defaults
     steps:
+      - checkout
       - attach_workspace:
           at: ~/project
       - run:
-          name: Test
-          command: yarn test --maxWorkers=2 --coverage && yarn codecov
-    environment:
-        - NODE_ENV: "test"
+          name: Tests
+          command: yarn test -w=2 --coverage && yarn coverage
 
+  # --------------------------------------------------------------------------
+  # Phase 4: Publish docs (master branch only)
+  # --------------------------------------------------------------------------
   docs:
     <<: *defaults
     steps:
+      - checkout
       - attach_workspace:
           at: ~/project
       - run:
-          name: Publish Docs
+          name: Publish docs
           command: yarn docs-publish
 
+# Workflow definition
 workflows:
   version: 2
-  setup-and-parallel-test:
+
+  # Build and test after each commit
+  commit:
     jobs:
       - setup
-
-      - lint:
-          requires:
-            - setup
-
-      - build:
-          requires:
-            - setup
-
-      - test:
-          requires:
-            - setup
-
-      - docs:
-          requires:
-            - setup
-          filters:
-            branches:
-              only:
-                - master
+      - lint:  { requires: [setup] }
+      - audit: { requires: [setup] }
+      - build: { requires: [setup] }
+      - lint:  { requires: [build] }
+      - test:  { requires: [build] }
```

Comment:
should this be `test-unit` instead of `test`?

Reply:
Renamed the job back to `test`.